client/core/account.go

```diff
@@ -2,11 +2,59 @@ package core
 
 import (
 	"encoding/hex"
+	"fmt"
 
 	"decred.org/dcrdex/client/db"
 	"github.com/decred/dcrd/dcrec/secp256k1/v3"
 )
 
+// AccountDisable is used to disable an account by given host and application
+// password.
+func (c *Core) AccountDisable(pw []byte, host string) error {
+	// Validate password.
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return codedError(passwordErr, err)
+	}
+
+	// Parse address.
+	addr, err := addrHost(host)
+	if err != nil {
+		return newError(addressParseErr, "error parsing address: %v", err)
+	}
+
+	// Get the dexConnection.
+	c.connMtx.RLock()
+	defer c.connMtx.RUnlock()
+	dc, found := c.conns[addr]
+	if !found {
+		return newError(unknownDEXErr, "DEX: %s", addr)
+	}
+
+	// Check active orders.
+	if dc.hasActiveOrders() {
+		return fmt.Errorf("cannot disable account with active orders")
+	}
+
+	acctInfo, err := c.db.Account(dc.acct.host)
+	if err != nil {
+		return newError(accountRetrieveErr, "Error retrieving account: %v", err)
+	}
+	err = c.db.DisableAccount(acctInfo)
+	if err != nil {
+		return newError(accountDisableErr, "Error disabling account: %v", err)
+	}
+	// Close dexConnection books.
+	for _, book := range dc.books {
+		book.close()
+	}
```

Comment:
Need to rethink this because the range on the `dc.books` map allows for a concurrent map write in another goroutine since this doesn't lock it's mutex.  However since `book.close` is actually `stopBook`, which locks the mutex too, we can't simply lock it here (this an example of why I don't love hiding the true implementations as opaque functions like `bookie.close`).

Maybe `dc.stopBook` for each market in `dc.cfg.Markets` (holding `dc.cfgMtx`), similar to in `handleReconnect`.  So, similar to what you had before https://github.com/decred/dcrdex/commit/fb916b134648b0faacdd3a38a28b3573462c09cc, but not ranging on `dc.marketMap()` but `dc.cfg.Markets` instead.